general/v0.1/protocol_basics.md

```diff
@@ -37,40 +36,93 @@ Communication between fediverse server and FASP MUST use HTTPS in production
 or production-like settings. This requirement MAY be relaxed in
 development environments.
 
-Registration tokens are JSON Web Tokens (JWT) as defined in
-[RFC-7519](https://datatracker.ietf.org/doc/html/rfc7519).
-
-For authentication and authorization of API calls, FASP and fediverse server
-use the OAuth 2.0 protocol as defined in
-[RFC-6749](https://tool.ietf.org/html/rfc6749.html).
-
 Custom API calls are HTTPS calls sending, if necessary, JSON data
 (`Content-Type: application/json`) and receiving JSON data.
 
-### OAuth2, Authentication and Authorization
+### Base URL
+
+As mentioned above both FASP and fediverse software MUST implement the
+API endpoints specified here relative to a base URL of their own
+choosing. This allows existing fediverse software and existing software
+projects that want to add the ability to act as FASP to implement this
+without confliciting with their existing API endpoints.
+
+To make the initial registrations of a FASP with a fediverse server
+easier, fediverse software MUST include their base URL as part of the
+`metadata` of their `nodeinfo` accessible via the `.well-known/nodeinfo`
+endpoint.
+
+
+Example `nodeinfo`:
+
+```json
+{
+  "version": "2.0",
+  "software": {
+    "name": "fediexample",
+    "version": "6.2.7"
+  },
+  "protocols": [
+    "activitypub"
+  ],
+  "services": {
+    "outbound": [],
+    "inbound": []
+  },
+  "openRegistrations": false,
+  "metadata": {
+    "nodeName": "fedi",
+    "faspBaseUrl": "https://fedi.example.com/fasp"
+  }
+}
+```
+
+### Request Integrity
+
+In order to allow both parties to verify the integrity of message
+contents, all requests MUST contain a `content-digest` HTTP header as
+defined by [RFC-9530](https://tools.ietf.org/html/rfc9530.html).
+
+The hashing algorith used MUST be SHA-256.
+
+### Authentication
 
 As described in [03: Registration](registration.md) both FASP and
-fediverse server use OAuth 2.0 to authorize API calls. Both MUST obtain a valid
-access token and send this as a "bearer token" in the `Authorization`
-HTTP header with every API call.
+fediverse server exchange client IDs and public keys. API requests are
+being authenticated by HTTP Message Signatures as defined in
+[RFC-9421](https://tools.ietf.org/html/rfc9421.html).
 
-Example header:
+The signature algorithm used is EdDSA Using Curve edwards25519.
+Signatures cover signature parameters, the derived components `@method`
+and `@target-uri` and the HTTP header `content-digest` (see previous
+section).
+
+The `keyid` parameter MUST include the client ID and the corresponding 
+private key is used to generate the signature.
```

Comment:
It might be better to write this as something like:

> During registration the FASP and Fediverse server both generate a unique public/private keypair and exchange the public keys with each other and an associated client identifier (client_id). When signing requests, the `keyid` paramter MUST be the client ID corresponding to the keypair generated during registration.